### PR TITLE
Update of dependencies and go

### PR DESCRIPTION
**Pull request by hoppea2**
# Description
Updated go to 1.17
Updated go in actions to 1.17
Updated 3rd party dependencies to latest
Updated Open Telemetry dependencies and code changes to Prometheus and Zipkin/tracing

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/127|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Executed regression test

### CODE REVIEW

**Comment by hoppea2** [comment]
run e2e test

**Comment by hoppea2** [comment]
run e2e test

**Review by shaynafinocchiaro — DISMISSED** [review]

File: internal/powerflex/token_getter.go [hunk 1]
```diff
@@ -75,7 +75,8 @@ func (tg *TokenGetter) Start(ctx context.Context) error {
 
 // GetToken returns a valid token for the configured PowerFlex
 func (tg *TokenGetter) GetToken(ctx context.Context) (string, error) {
-	ctx, span := trace.SpanFromContext(ctx).Tracer().Start(ctx, "GetToken")
+	//ctx, span := trace.SpanFromContext(ctx).Tracer().Start(ctx, "GetToken")
```
Comment (atye) [inline internal/powerflex/token_getter.go:78]: Should this be removed?

**Review by atye — APPROVED** [review]

File: cmd/proxy-server/main.go [hunk 2]
```diff
@@ -238,7 +243,19 @@ func run(log *logrus.Entry) error {
...
 
-	metricsExp, err := prometheus.InstallNewPipeline(prometheus.Config{})
+	config := prometheus.Config{}
+	c := controller.New(
+		processor.NewFactory(
+			selector.NewWithHistogramDistribution(
+				histogram.WithExplicitBoundaries(config.DefaultHistogramBoundaries),
+			),
+			aggregation.CumulativeTemporalitySelector(),
+			//export.CumulativeExportKindSelector(),
```
[the remark this answers is not in the document]
Comment (hoppea2) [reply cmd/proxy-server/main.go]: fixed

File: internal/proxy/root_handler.go [hunk 3]
```diff
@@ -48,7 +48,7 @@ func (h *RootHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.next.ServeHTTP(w, r)
 	//	h.once.Do(func() {
 	//		h.meter = otel.Meter("karavi/count")
-	//		h.key = label.Key("path").String("/")
+	//		h.key = attribute.Key("path").String("/")
```
Comment (shaynafinocchiaro) [inline internal/proxy/root_handler.go:51]: Do we want to remove these comments as well?

**Review by atye — APPROVED** [review]

**Review by alikdell — APPROVED** [review]

**Pull request merged**